app/Services/Manager.php

```diff
@@ -80,4 +97,62 @@ public function collectContestInfo($data)
 
         return $contest;
     }
+
+    /**
+     * Get user signup for a specific campaign and run. If only user id is
+     * provided, the send back all user signups.
+     *
+     * @param  string $id
+     * @param  string $campaign    Campaign ID
+     * @param  string $campaignRun Campaign Run ID
+     *
+     * @return object $signups
+     */
+    public function getUserSignup($id, $campaign = null, $campaignRun = null)
+    {
+        $signups = $this->northstar->getUserSignups($id, $campaign);
+
+        // Only return the sign up record for the run that was specified.
+        if ($campaignRun) {
+            foreach ($signups as $key => $signup) {
+                if ($signup->campaign_run->id == $campaignRun) {
```

Comment:
@weerd i was having some issues with this because the campaign run id that comes back from the northstar call is an integer, and the campaign run id that comes back from the gladiator db is a string.